EventKit/eventkit-cloud
# Refactor DataPackShareDialog to functional component

# Description of Improvement

Refactors DataPackShareDialog to use functional component, and updates tests to use react testing library.

## How to test

Create a datapack
view on status page
Modify permissions via dialog
Verify dialog still behaves as expected
Verify tests pass

## Quality Assurance

The following items have been updated:


## Screenshots



:smile:
